reflex/app.py

```diff
@@ -112,11 +112,29 @@ def default_backend_exception_handler(exception: Exception) -> EventSpec:
         EventSpec: The window alert event.
 
     """
+    from reflex.components.sonner.toast import Toaster, toast
+
     error = traceback.format_exc()
 
     console.error(f"[Reflex Backend Exception]\n {error}\n")
 
-    return window_alert("An error occurred. See logs for details.")
+    error_message = (
+        [f"{type(exception).__name__}: {exception}.", "See logs for details."]
+        if get_config().debug is True
+        else ["Contact the website administrator."]
+    )
+    if Toaster.is_used:
```

Comment:
This doesn't seem to work for me out of the box? I have to do `rx.toast.provider()` within my app and then it shows up

Reply:
Fixed this by making sure that `default_overlay_component` is always called before any page compilation.